### DISCUSSION TITLE
Multi-UAV Path Planning for Urban Air Mobility (Project 247)

### DISCUSSION THREAD

**Discussion by robertogl** [mathworks/MATLAB-Simulink-Challenge-Project-Hub#85] [Collaborate]
Contribute to the discussion by asking and/or answering questions, commenting, or sharing your ideas for solutions to [project 247](https://github.com/mathworks/MathWorks-Excellence-in-Innovation/blob/main/projects/Multi-UAV%20Path%20Planning%20for%20Urban%20Air%20Mobility)

---
*:robot: An AI bot may reply to your comments with suggestions and guidance. It will not solve the project for you. Your comments may be processed by a third-party LLM. To direct a question to a specific person, tag them with @username.*

**Comment by aerosagar** [comment]
Hi, I am a student at Dayananda Sagar College of Engineering and interested in working and collaborating on this project

**Comment by KISHANCJOLAPARA** [comment]
Hey, I am an undergrad student at Dayananda Sagar College of  Engineering with an interest in Aerial robotics. I'm open to and would love to work in collaboration on this project!

**Comment by leebissessar5** [comment]
Hey there, I'm a graduate engineering student and have some experience with UAVs and various other autonomous systems, specifically in simulation. I have recently picked up an interest in swarm robotics and this project seems like a good candidate to get started. I'm open to collaborating with others!

**Comment by manohar2028** [comment]
Hi, I am a student at GCOEARA and interested in working and collaborating on this project. 
I have some doubts. Actually what is supposed to be done. Either we have to make model simulation/ animation or just algorithm.

**Reply by robertogl** [reply]
It would be great to have some sort of visualization to show that the algorithm is effective. It is up to you how advanced you make it. However, I think you most likely will need to consider the UAV model for the algorithm.

**Reply by manohar2028** [reply]
Ok. Thank you Sir for clarification.

**Comment by rdlynx19** [comment]
Hey, I am an undergrad student at VNIT, Nagpur, India and am interested in working and collaborating on this project.

**Comment by aerosagar** [comment]
Hello,
I am working on the project, however, the deadline is to approach in 2 weeks. I have mailed the team seeking an extension. Have not heard back from them since.
Please help me with the extension process.
Thanks

**Reply by robertogl** [reply]
Sorry I have been late in my replies. Yes, you can have an extension. Let me find your email and reply to that, but do not worry about the upcoming deadline.

Thanks.

**Reply by aerosagar** [reply]
Thanks!

**Comment by rdlynx19** [comment]
We are using the LIDAR and RADAR fusion example in MATLAB and trying to implement RRT planning algorithm. For this we need a 3D occupancy map to be generated but are unable to do so on a scenario generated from the UAV Designer App. Does anyone have any suggestions for the same?

**Reply by aerosagar** [reply]
https://in.mathworks.com/help/uav/ug/generate-random-3-d-occupancy-map-for-uav-motion-planning.html

**Reply by aerosagar** [reply]
Hope this link helps.
I do have a doubt too, in one of the examples, the osm file of Manhattan city is used and the map is brought into simulation. My doubt is, how do we introduce the drone constraints into this.

**Reply by lovesh1711** [reply]
Hello,
Actually, we looked into this documentation but it focuses on generating occupancy grids from scratch. But we are wondering if it's possible to directly make a 3D occupancy map from a custom map made by the UAV designer app.

Regarding adding drone constraints, we were thinking of taking the reference of the ExampleHelperUAVStateSpace function given in "Motion planning with RRT for a fixed-wing UAV example"

**Reply by aerosagar** [reply]
Thanks.
If I get a lead, I will put it up in this thread.

**Reply by robertogl** [reply]
Hi all,

Unfortunately there is no direct method to get an occupancy grid from the UAV Designer app. As you probably know, you can build the map using a lidar sensor. We have an example [here](https://www.mathworks.com/help/uav/ug/map-environment-motion-planning-using-uav-lidar.html).

I hope that helps.

**Comment by abbasfadavi** [comment]
Hello to all
In this link, I implement a simple project for route planning
Thank you for your comments.  
https://github.com/abbasfadavi/Multi-Path-Planning

**Comment by azeem-sarwar** [comment]
Hi, I am a student at Nordhausen university of applied sciences and interested in working and collaborating on the project 247.

**Comment by InfiniteBoyEric** [comment]
Hello, I am a senior student at Shandong University and interested in working and collaborating on this project. I am looking forward to having discussions with you all, and get assistance just in case.

**Comment by manohar2028** [comment]
Yes sure. Where should we connect ?

**Reply by InfiniteBoyEric** [reply]
Is it convenient for you to contact me by email, or could I have your email please? My email is zengjiaxiangboy@163.com, and I am looking forward to having an in-depth communication with you. Or do you have any recommendations in connection?
BTW, I think I am a green learner and it seems that maybe there are many problems I would ask you experienced people.(LOL)

**Comment by Ebi25** [comment]
Hello,i have sumbitted the solution it's been 2 weeks i have'nt recived any mail from your side other than confirmation mail,i would like to know status of the submission.

**Reply by robertogl** [reply]
Working on it.

**Comment by bteodora** [comment]
Hello, everyone! I’m a student at Faculty of Technical Sciences in Novi Sad and I’ve been working on this project for some time now. Realized I never actually said hi in the discussion thread – better late than never, I guess! Looking forward to seeing how everyone's projects turn out and to possibly collaborate in the future!
